regression-test/suites/variant_p0/insert_into_select.groovy

```diff
@@ -55,4 +55,189 @@ suite("regression_test_variant_insert_into_select", "variant_type"){
+        ),
+        (7,
+          ['p1', 'q1', 'r1', 's1', 't1'],
+          map(100, 'p1', 200, 'q1', 300, 'r1', 400, 's1', 500, 't1'),
+          named_struct('a', 5, 'b', 'b5', 'c', '192.168.1.5', 'd', 5.0),
+          '{"info": "random text", "count": 5}',
+          array(named_struct('a', 5, 'b', 'b5', 'd', '2021-05-01 00:00:00')),
+          {},
+          named_struct('a', [5], 'b', map(3, 'r5'), 'c', named_struct('a', 5, 'b', 'b5', 'c', '192.168.1.5', 'd', 5.0))
+        );
+        """
+        // select the origin table
+        qt_sql """select * from ${table_name}_complex order by k"""
+
+        // then insert into select from ${table_name}_complex to ${table_name}_var with key increment by 1
+        sql """insert into ${table_name}_var select k, a from ${table_name}_complex"""
+        sql """insert into ${table_name}_var select k+7, b from ${table_name}_complex"""
+        sql """insert into ${table_name}_var select k+14, c from ${table_name}_complex"""
+        sql """insert into ${table_name}_var select k+21, d from ${table_name}_complex"""
+        sql """insert into ${table_name}_var select k+28, a_s from ${table_name}_complex"""
+        sql """insert into ${table_name}_var select k+35, b_s from ${table_name}_complex"""
+        sql """insert into ${table_name}_var select k+42, c_s from ${table_name}_complex"""
+        qt_sql """select * from ${table_name}_var order by k"""
+
+    def load_json_data = {tn, file_name ->
+        // load the json data
+        streamLoad {
+            table """$tn"""
+
+            // set http request header params
+            set 'read_json_by_line', 'true'
+            set 'jsonpaths', '[\"$.v.id\", \"$.v.type\", \"$.v.actor\", \"$.v.repo\", \"$.v.payload\", \"$.v.public\", \"$.v.created_at\"]'
+            set 'format', 'json'
+            set 'max_filter_ratio', '0.1'
+            file file_name // import json file
+            time 10000 // limit inflight 10s
+
+            // if declared a check callback, the default check condition will ignore.
+            // So you must check all condition
+
+            check { result, exception, startTime, endTime ->
+                if (exception != null) {
+                    throw exception
+                }
+                logger.info("Stream load ${file_name} result: ${result}".toString())
+                def json = parseJson(result)
+                assertEquals("success", json.Status.toLowerCase())
+                // assertEquals(json.NumberTotalRows, json.NumberLoadedRows + json.NumberUnselectedRows)
+                assertTrue(json.NumberLoadedRows > 0 && json.LoadBytes > 0)
+            }
+        }
+    }
+
+        def table_name_j = "github_events_j"
+        sql """DROP TABLE IF EXISTS ${table_name_j}"""
+        sql """
+            CREATE TABLE `${table_name_j}` (
+              `id` BIGINT NOT NULL,
```

Comment:
1. add schema change case (double write during schema change, like double_write_schema_change)
2. add literal cast like cast(cast('{"a" : 1}' as json) as variant) also, for map, struct
3. add large data set to schema_change_p2 to test stability